Shift+RightClick creates waypoints on Ore Samples

Shananiganeer opened the pull request:
Ore samples fire PlaceWaypointEvent when shift-rightclicked, closes #86.

Shananiganeer committed:
Add getBool

Added getBool method to facilitate the requireTool value for SubBlockOreSamplePart.

Shananiganeer committed:
Add requireTool PartDataPiece

Added (boolean) requireTool to control whether ore samples require appropriate tools.

Shananiganeer committed:
Add requireTool DataValue

(boolean) requireTool: if true, sets the return value of getMaterial() to Material.STONE and prevents onBlockActivated() from dropping items

Shananiganeer committed:
fix the oopsies

typos and prevent 'falling' samples from spawning drops if requireTool = true.

Shananiganeer committed:
OrePart bugfix

OreParts that were generated with variants would leave an orphaned non-variant MaterialPart that did not have an associated SubBlock and would cause a runtime error if getItemStack() was called.

Shananiganeer committed:
Fix single line {}s

Shananiganeer committed:
Merge branch 'develop/1.12.0' of https://github.com/The-Acronym-Coders/BASE into develop/1.12.0

Shananiganeer committed:
PR changes

Removed unnecessary key verification from removeMaterialPart().
Changed SubBlockOreSamplePart.onNeighborChange() to use dropBlockAsItem() per #93.

Shananiganeer committed:
Merge branch 'develop/1.12.0' of https://github.com/The-Acronym-Coders/BASE into develop/1.12.0

Shananiganeer committed:
Ore Samples create waypoints

Created PlaceWaypointEvent with constructor:
  PlaceWaypointEvent(
    String modId,
    String id,
    String name,
    int dimension,
    BlockPos pos)

During initialization, JourneyMapModule registers
JourneyMapWaypointListener to listen for PlaceWaypointEvent and creates
a random colored JourneyMap waypoint at event.pos.
When shift+right clicked, ore samples post PlaceWaypointEvent.

SkySom commented on src/main/java/com/teamacronymcoders/base/modules/journeymap/JourneyMapModule.java:
```diff
@@ -0,0 +1,44 @@
+
+@Module(MODID)
+@journeymap.client.api.ClientPlugin
+public class JourneyMapModule extends ModuleBase implements IClientPlugin {
```
This will crash without JourneyMap installed, no?

SkySom commented on src/main/java/com/teamacronymcoders/base/modules/journeymap/JourneyMapWaypointListener.java:
```diff
@@ -0,0 +1,27 @@
+        try {
+            jAPI.show(new Waypoint(event.getModId(), event.getId(), event.getName(), event.getDimension(), event.getPos()).setColor(new Random().nextInt(0xffffff)));
+        } catch(Throwable t) {
+            instance.getLogger().warning(t.getLocalizedMessage());
```
Don't import static this.

Shananiganeer committed:
Changes for PR

Seperate JourneyMap plugin code from JourneyMap module code.
Remove import static for non-final fields.
Change Waypoint ID to include dimension and sample name.
IDs were not unique enough and would be identical if samples in
different dimensions were at the same X, Y, Z.

Shananiganeer committed:
Make event more generic; Respect module config

Removed modid and id fields from PlaceWaypointEvent.
JoureyMapWaypointListener generates the appropriate modId and id for
the Waypoint constructor instead of pulling from PlaceWaypointEvent.
Listener is only registered if B:"JourneyMap enabled"=true in config.

SkySom commented on src/main/java/com/teamacronymcoders/base/modules/journeymap/JourneyMapPlugin.java:
```diff
@@ -0,0 +1,29 @@
+import static com.teamacronymcoders.base.Reference.MODID;
+import static net.minecraftforge.common.MinecraftForge.EVENT_BUS;
+
+@journeymap.client.api.ClientPlugin
```
Any specific reason why this has the fully qualified name?

Shananiganeer replied:
Nope, I was following the JMApi example by rote instead of using my brain.

SkySom commented on src/main/java/com/teamacronymcoders/base/modules/journeymap/JourneyMapWaypointListener.java:
```diff
@@ -0,0 +1,28 @@
+    @SubscribeEvent
+    public void onWaypointPlaced(PlaceWaypointEvent e) {
+        try {
+            jAPI.show(new Waypoint(MODID, e.getName()+e.getDim()+e.getPos().hashCode(), e.getName(), e.getDim(), e.getPos()).setColor(new Random().nextInt(0xffffff)));
```
Can just pass color.

SkySom replied:
Also spaces between + :D

Shananiganeer committed:
Pass material color to waypoint; Code cleanup

Added color field to PlaceWaypointEvent.
SubBlockOrePart posts waypoint event with the color of it's material.
Import ClientPlugin istead of linking the fully qualified name.
Fixed spacing around +s.
